src/GraphQL/Validation/Complexity/ComplexityAnalyzer.cs

```diff
@@ -140,6 +142,9 @@ private static GraphQLFragmentDefinition GetFirstFragmentWithoutPendingDependenc
                     {
                         if (selection is GraphQLFragmentSpread spread)
                         {
+                            if (!_visitedFragments.Add(spread.FragmentName.Name.Value))
+                                throw new NoFragmentCyclesError(document, spread.FragmentName.Name.StringValue);
```

Comment:
I didn’t review all the code here, but my gut feeling is that to match all of our other code this should not this an error, but silently ignore the situation. There are many situations which can break the complexity analyzer (or similarly the authorization rule), such as referencing a type, fragment or variable that does not exist.

Reply:
In v8, when we implement our plans to have built-in rules run first, then read arguments, then other rules, then we do not need to worry about these kinds of situations here, or in the authorization rule, or within third-party rules, since the built-in rules would prevent such a situation from occurring. Sorry I have not worked on it recently, but I’ll get it done.

Reply:
I changed this check to the primitive one. It works for now.